baremetal/metal3machine_manager.go

```diff
@@ -1323,8 +1323,10 @@ func (m *MachineManager) GetProviderIDAndBMHID() (string, *string) {
 	// instead contains / to separate the names. In that case we return nil for
 	// the bmh ID to force the controller to fetch it differently.
 	if strings.Contains(bmhID, "/") {
+		m.Log.Info("ProviderID is in new format, it does not contain the BMH ID", "providerID", *providerID)
 		return *providerID, nil
 	}
+	m.Log.Info("ProviderID contains the BMH ID", "providerID", *providerID)
```

Comment:
I think this could be a bit too spammy at the Info level. If I'm not mistaken this would be logged on each reconciliation loop. I suggest changing it to Debug level instead.

Reply:
I'm not very familiar with the architecture of this piece of code, not sure if it should be modified like this. 
@mboukhalfa @furkatgofurov7 Can you take a look at this suggestion?

Reply:
Hmm this is called every time in the normal reconcile function https://github.com/metal3-io/cluster-api-provider-metal3/blob/6f5bcd190129905fcc657d0f8b2fc5f227c1c467/controllers/metal3machine_controller.go#L265

Agree let's make it debug level